pkg/cloudprovider/types.go

```diff
@@ -373,3 +373,24 @@ func IsNodeClassNotReadyError(err error) bool {
 	var nrError *NodeClassNotReadyError
 	return errors.As(err, &nrError)
 }
+
+// InstanceCreateError is an error type returned by CloudProviders when instance creation fails
+type InstanceCreateError struct {
```

Comment:
I wonder if we should just call it `Error` or `ConditionError` or `PrettyError`. Realistically, I don't think we have to be specific and call this a create error.

Reply:
Addressed